### ISSUE TITLE
Update docs for optional extras

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8560]
|[<img alt="derwolfe's avatar" src="https://avatars.githubusercontent.com/u/641357?s=50" width="50" height="50">](https://github.com/derwolfe)| @derwolfe reported|
|-|-|
|Trac ID|trac#8560|
|Type|enhancement|
|Created|2016-07-04 15:38:27Z|

ticket:8483 added a warning to the documentation to make it clear that optional extras were not functional on python3, this was recently fixed by ticket:7807.  However, this change did make it clear that only certain `dev` dependencies are available on python3 [https://github.com/twisted/twisted/blob/trunk/twisted/python/dist.py#L70]

As such, the narrative docs should be updated to include
1. New documentation for the `dev` option on python2
2. Removal of the warning about optional extras not being available on python3.
3. It looks like there are new options for http2 [https://github.com/twisted/twisted/blob/trunk/twisted/python/dist.py#L107], this could be taken care of by another ticket or this one, but these should be documented as well. 

Thanks!

<details><summary>Searchable metadata</summary>

```
trac-id__8560 8560
type__enhancement enhancement
reporter__derwolfe derwolfe
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1467646707140871 1467646707140871
changetime__1474198049187875 1474198049187875
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="derwolfe's avatar" src="https://avatars.githubusercontent.com/u/641357?s=50" width="50" height="50">](https://github.com/derwolfe)<a name="note_1"></a>|@derwolfe removed owner|
|-|-|

PR available here [https://github.com/twisted/twisted/pull/292]

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset a36b8f3bbd970b6cf3726be3807672e09eb0db59
```
#!CommitTicketReference repository="" revision="a36b8f3bbd970b6cf3726be3807672e09eb0db59"
Merge pull request #293 from twisted/8560-update-docs-extras

Author: derwolfe
Reviewer: rodrigc
Fixes: #8560

Update documentation for optional extras
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
